lib/application.js

```diff
@@ -471,15 +539,47 @@ var RequireAuthorizationApplication = Application.extend({
             });
         });
     },
-    getAccessToken: function(token, secret, verifier, callback, options) {
+    setAccessToken: function(token, secret, verifier, callback, options) {
         var self = this;
 
         return new Promise(function(resolve, reject) {
-            self.oa.getOAuthAccessToken(token, secret, verifier, function(err, token, secret, results) {
+            self.oa.getOAuthAccessToken(token, secret, verifier, function(err, results) {
                 if (err)
                     reject(err);
-                else
-                    resolve({ token: token, secret: secret, results: results });
+                else {
+                    var exp = new Date();
+                    exp.setTime(exp.getTime() + (results.oauth_expires_in * 1000));
+                    self.setOptions({
+                        accessToken: results.oauth_token,
+                        accessSecret: results.oauth_token_secret,
+                        sessionHandle: results.oauth_session_handle,
+                        tokenExpiry: exp.toString()
```

Comment:
I usually prefer to use `toISOString()` if it's not going to a human/UI. It won't matter for this code, but this token will usually end up in a database once it's returned. There are all kinds of databases out there and most of them suck at dates. Using an ISO string gives our little token the best chance at being serialised correctly and living a full and happy life.